### DISCUSSION TITLE
Support for disabling the tab bar entirely

### DISCUSSION THREAD

**Discussion by RobertCraigie** [ghostty-org/ghostty#2759] [Feature Requests, Ideas] [answered]
I like having my terminal as minimal as possible so I'd really enjoy hiding the tab bar and then relying on cycling through tabs or using the dedicated tab viewer to switch tabs.

Right now it looks like you can hide the tab bar when there is only one tab open

![Screenshot 2024-11-21 at 15 37 54](https://github.com/user-attachments/assets/56fdd2ce-00c3-4f72-b833-3325d44009d5)

![Screenshot 2024-11-21 at 15 38 13](https://github.com/user-attachments/assets/67cdaf8e-4314-4ae8-91d7-8515787b2472)

---

I'm coming from WezTerm where this is [supported](https://wezfurlong.org/wezterm/config/lua/config/enable_tab_bar.html) and the active tab number & total number of tabs is included in the title bar.

![Screenshot 2024-11-21 at 15 40 23](https://github.com/user-attachments/assets/5d146ebc-e201-4d06-9a9a-84cf39174492)

**Comment by mitchellh** [comment, marked as answer]
I think this was requested in another discussion. If we find that we can close this or that discussion. In any case, I'm fine supporting this _if the native GUI toolkit supports it_. WezTerm doesn't use native tabs, so they have full control to do whatever they want. We're beholden by what the underlying OS provides. 

I think macOS native tabs do not allow this when there is more than one tab, but I could be wrong on that point.

**Reply by RobertCraigie** [reply]
Thanks!

> I think macOS native tabs do not allow this when there is more than one tab, but I could be wrong on that point.

ahhhh I think you're right, that's a shame :(

**Comment by hustcer** [comment]
I'm a fun of this feature
